src/wallet/wallet_rpc_server.cpp

```diff
@@ -3155,7 +3155,21 @@ namespace tools
       er.message = "Failed to create wallet";
       return false;
     }
-    wal->set_seed_language(req.language);
+    if (!req.language.empty())
```

Comment:
You're duplicating code from just a few lines up, why? (lines 3122-... here)

Reply:
Since language validation appears to be done in at least 3 places, maybe this chunk should be moved into its own function.

Reply:
Because I missed it as it was done earlier than I looked.